RNWCPP/ReactUWP/Views/ReactControl.h

```diff
@@ -72,6 +74,7 @@ class ReactControl
 
   winrt::Grid m_redBoxGrid { nullptr };
   winrt::TextBlock m_errorTextBlock { nullptr };
+  WeakThis<ReactControl> m_weakThis{ this };
```

Comment:
>WeakThis<ReactControl> m_weakThis{ this }; [](start = 2, length = 42)

This is confusing me.  It looks like this actually takes a strong reference on this in the constructor, not a weak reference.
This appears to create a cyclic reference and I don't ever see the cycle broken.  Isn't this causing ReactControl to leak?

Reply:
Look like WeakThis and m_weakThis is not a good name for it.
I rename it to ThisWrapper and m_thisWrapper.
It's not a strong reference/weak reference to this, but just a wrapper of this to help to get weak_ref of This and resolve it to This, then resolve to this.
more details see ThisWrapper.h